README.md

```diff
@@ -87,14 +87,20 @@ Selecting the format you'd like your project to be exported as while choosing th
 
 ![Alt Text](https://media.giphy.com/media/I5g06mUnVzdX7iT2Gf/giphy.gif)
 
-## Deving locally
-1. clone repo
-2. create a virtual env: `virtualenv local_dev`
-3. activate virtual env: `source local_dev/bin/activate`
-4. install packages from repo: `pip3 install -e ".[dev]"`
-5. create script on top level of `ROBOFLOW-PYTHON` and reference local `roboflow` directory: `from roboflow import Roboflow`
-6. when done, uptick the pip package minor version number in `setup.py`
-7. manually add any new dependencies to the `requirements.txt` and list of dependencies in `setup.py` (careful not to overwrite any packages that might screw up backwards dependencies for object detection, etc.) 
+## Developing locally
+
+1. clone this repo
+1. with docker
+    1. `docker build -t roboflow`
+    1. copy `.env-example` to `.env` and update environment variables for testing
+    1. shell into container: `docker run -it --rm --env-file .env -v /your/path/to/roboflow-python:/roboflow roboflow`
+1. with virtualenv
+    1. create a virtual env: `virtualenv local_dev`
+    1. activate virtual env: `source local_dev/bin/activate`
+    1. install packages from repo: `pip3 install -e ".[dev]"`
+1. create a script on the top level `ROBOFLOW-PYTHON` directory and reference local `roboflow` directory: `from roboflow import Roboflow`
+1. when done, uptick the pip package minor version number in `setup.py`
+1. manually add any new dependencies to the `requirements.txt` and list of dependencies in `setup.py` (careful not to overwrite any packages that might screw up backwards dependencies for object detection, etc.) 
```

Comment:
To develop locally

### Virtual Env

Be sure to pass over the correct environments variables, see [Tests](### Tests) section

```bash
# clone repo and cd inside it
git clone git@github.com:roboflow-ai/roboflow-python.git && cd roboflow-python
# create venv
virtualenv .venv 
# activate it
source .venv/bin/activate 
# install library in edit mode
pip install -e ".[dev]"
# run test to check everything ok
python -m unittest
```

### Docker

```bash
# clone repo and cd inside it
git clone git@github.com:roboflow-ai/roboflow-python.git && cd roboflow-python
# run docker python 3.8 container and map current dir to /roboflow
docker run -it -v $(pwd):/roboflow python:3.8.14 /bin/bash
# install library in edit mode
cd /roboflow && pip install -e ".[dev]"
# run test to check everything ok
python -m unittest
```

Reply:
🙏 your documentation format is definitely more clear than mine.

the `Dockerfile` is certainly not necessary, but one of the reasons that i like it is because it comes with defined consistency. it also copies and installs dependencies upon initial build.

with the suggested approach of not using a `Dockerfile`, every time you start the container, you have to manually CD into the project's root directory and reinstall dependencies before you can use the application.

i also chose `python:3.6-slim` because `3.6` is the minimum version supported in `setup.py` and the `slim` image is almost 8 times smaller than the `python:3.8.14` image...

i'm curious about your thoughts here!